textile/controllers/textile_pricing_rule.py

```diff
@@ -226,6 +229,14 @@ def sorting_function(d):
 	def get_customer_base_rate(cls, customer):
 		return None
 
+	@classmethod
+	def is_fixed_base_rate(cls, customer):
+		if cls.doctype == "Pretreatment Pricing Rule":
+			return frappe.get_cached_value("Customer", customer, "is_fixed_pretreatment_rate")
+		if cls.doctype == "Print Pricing Rule":
+			return frappe.get_cached_value("Customer", customer, "is_fixed_printing_rate")
+		return False
+
```

Comment:
This should be an abstract method like we discussed!
Implementation should be in PrintPricingRule and PretreatmentPricingRule